### PR TITLE
NEW Migrate code from other modules

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/.github/issues/342

Separate commits so please do no squash merge

Code copied from silverstripe/securityreport has had any code relating to subsites removed

Any code that was relating to silverstripe/subsites or silverstripe/contentreport has been removed as those module will not be commercially supported going forward

### CODE REVIEW

File: tests/behat/src/FeatureContext.php
Comment (GuySartorelli) [inline tests/behat/src/FeatureContext.php:1]: Need composer autoload entry for `tests/behat/src`

File: client/src/styles/sitewidecontentreport.scss
Comment (GuySartorelli) [inline client/src/styles/sitewidecontentreport.scss:1]: If we're having stuff in `client/src` and `client/dist` then the `javacscript` dir should be included and built out at the same time.

It's fair if you want to say that's out of scope (though ideally it'd be done at the same time as we update the folder structure, and will take very little time to do) - in that case it needs to have a follow-up card.

File: README.md [hunk 1]
```diff
@@ -114,6 +117,102 @@ SilverStripe\Reports\SitewideContentReport\SitewideContentReport:
     - SitewideContentReportQueuedExportExtension
 ```
 
+## External broken links report
```
Comment (GuySartorelli) [inline README.md]: These should Moved to the main docs, or else this module's docs should be exposed in the "Optional features" section.

Given reports are part of installer, probably the former.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
The external broken links report doesn't work. I click "create new report" and it does nothing. Looking at the dev console there are some missing files. I think you're missing the `expose` entry for `client/dist` in `composer.json`

The folder structure between the newly copied code is also not standardised. e.g. `ExternalLinks` has a `Reports` subfolder, but the other two don't. Please make the folder structure uniform across all three feature dirs.

File: tests/behat/src/FeatureContext.php
Comment (emteknetnz) [reply tests/behat/src/FeatureContext.php:1]: That's already in the PR?

Comment (GuySartorelli) [reply tests/behat/src/FeatureContext.php:1]: So it is 😅

File: client/src/styles/sitewidecontentreport.scss
Comment (emteknetnz) [reply client/src/styles/sitewidecontentreport.scss:1]: Done

File: README.md (hunk 1 above)
Comment (emteknetnz) [reply README.md]: The problem there is we'd basically need to a new reports section. Currently the only reports thing there is https://docs.silverstripe.org/en/5/developer_guides/customising_the_admin_interface/how_tos/customise_site_reports/ which doesn't seem like an appropriate place to dump in info about the external links report etc

We're currently just moving docs from one readme to another, I think at least for the scope of what we're doing this is acceptable

**Comment by emteknetnz** [comment]
> The external broken links report doesn't work. I click "create new report" and it does nothing. Looking at the dev console there are some missing files. I think you're missing the expose entry for client/dist in composer.json

Works fine on my local

> The folder structure between the newly copied code is also not standardised. e.g. ExternalLinks has a Reports subfolder, but the other two don't. Please make the folder structure uniform across all three feature dirs.

Have updated

File: code/SiteWideContentReport/Model/SitewideContentTaxonomy.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace SilverStripe\Reports\SiteWideContentReport\Model;
```
Comment (GuySartorelli) [inline code/SiteWideContentReport/Model/SitewideContentTaxonomy.php]:
```suggestion
namespace SilverStripe\Reports\SiteWideContentReport\Extensions;
```
And rename the `Model` dir to `Extensions`

File: README.md (hunk 1 above)
Comment (GuySartorelli) [reply README.md]: I really don't think documentation belongs in a readme. At the very least move it into a `docs/` dir in here.

**Comment by GuySartorelli** [comment]
~~External broken links report still doesn't work for me. I'm just using installer (not sink) and pulling in only this PR since it shouldn't rely on any of the other changes from what I can tell.~~

Ignore me, it works fine, I just was pulling it in wrong

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: code/SiteWideContentReport/Model/SitewideContentTaxonomy.php (hunk 2 above)
Comment (emteknetnz) [reply code/SiteWideContentReport/Model/SitewideContentTaxonomy.php]: Done

**Comment by emteknetnz** [comment]
Have moved docs for external links + site wide content report do docs/en/index.md

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**